stixcore/products/level0/quicklook.py

```diff
@@ -0,0 +1,207 @@
+        control['compression_scheme_counts_skm'].meta = {'NIXS': ['NIXD0101', 'NIXD0102',
+                                                                  'NIXD0103']}
+        control['compression_scheme_triggers_skm'] = \
+            _get_compression_scheme(packets, 'NIXD0104', 'NIXD0105', 'NIXD0106')
+        control['compression_scheme_triggers_skm'].meta = {'NIXS': ['NIXD0104', 'NIXD0105',
+                                                                    'NIXD0106']}
+        control['num_energies'] = _get_num_energies(packets)
+        control['num_energies'].meta = {'NIXS': 'NIX00270'}
+        control['num_samples'] = np.array(packets.get('NIX00271')).flatten()[
+            np.cumsum(control['num_energies']) - 1]
+        control['num_samples'].meta = {'NIXS': 'NIX00271'}
+
+        time, duration = control._get_time()
+        # Map a given entry back to the control info through index
+        control_indices = np.hstack([np.full(ns, cind) for ns, cind in
+                                     control[['num_samples', 'index']]])
+
+        # cs, ck, cm = control['compression_scheme_counts_skm'][0]
+        counts = np.hstack(packets.get('NIX00272'))
+        # counts, counts_var = decompress(counts, s=cs, k=ck, m=cm, return_variance=True)
+
+        # ts, tk, tm = control['compression_scheme_triggers_skm'][0]
+        triggers = np.hstack(packets.get('NIX00274'))
+        # triggers, triggers_var = decompress(triggers, s=ts, k=tk, m=tm, return_variance=True)
+        # this may no longer be needed
+        # flat_indices = np.hstack((0, np.cumsum([*control['num_samples']]) *
+        #                           control['num_energies'])).astype(int)
+
+        # counts_reformed = np.hstack(c for c in counts).T
+        # counts_var_reformed = np.hstack(c for c in counts_var).T
+        # counts_reformed = [
+        #     np.array(counts[flat_indices[i]:flat_indices[i + 1]]).reshape(n_eng, n_sam)
+        #     for i, (n_sam, n_eng) in enumerate(control[['num_samples', 'num_energies']])]
+
+        # counts_var_reformed = [
+        #     np.array(counts_var[flat_indices[i]:flat_indices[i + 1]]).reshape(n_eng, n_sam)
+        #     for i, (n_sam, n_eng) in enumerate(control[['num_samples', 'num_energies']])]
+
+        # counts = np.hstack(counts_reformed).T
+        # counts_var = np.hstack(counts_var_reformed).T
+
+        data = Data()
+        data['control_index'] = control_indices
+        data['time'] = time
+        data['timedel'] = duration
+        data['triggers'] = triggers
+        data['triggers'].meta = {'NIXS': 'NIX00274'}
+        # data['triggers_err'] = np.sqrt(triggers_var)
+        data['rcr'] = np.hstack(packets.get('NIX00276')).flatten()
+        data['rcr'].meta = {'NIXS': 'NIX00276'}
+        data['counts'] = counts.T
+        data['counts'].meta = {'NIXS': 'NIX00272'}
+        # data['counts_err'] = np.sqrt(counts_var).T * u.ct
```

Comment:
where is the energy (axis/dimension)

Reply:
Well counts is time x detector x pixel x energy channel but the axis isn't added until l1 at the moment.